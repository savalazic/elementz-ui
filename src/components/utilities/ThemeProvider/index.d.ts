export { default, ThemeProps } from './ThemeProvider';
