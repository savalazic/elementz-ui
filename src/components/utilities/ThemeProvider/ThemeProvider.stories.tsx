import React from 'react';

import { storiesOf } from '@storybook/react';

import { infoConfig } from '../../../utils/info-config';
import { ThemeProvider } from './ThemeProvider';

import { Button } from '../../elements';

const stories = storiesOf('Utilities|ThemeProvider', module);

stories.addParameters({
  info: {
    ...infoConfig.info,
    propTables: [ThemeProvider],
    propTablesExclude: [Button],
  },
});

stories.add('default', () => {
  const theme = {
    primary: 'blue',
  };

  return (
    <ThemeProvider theme={theme}>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Et similique
        labore eaque, officiis accusantium veniam voluptatibus libero. Sapiente
        optio voluptas accusamus. Qui ipsum recusandae maxime odio magnam
        delectus provident nisi?
      </p>
      <Button>Button</Button>
      <Button type="primary">Button</Button>
      <Button ghost disabled>
        Button
      </Button>
    </ThemeProvider>
  );
});
