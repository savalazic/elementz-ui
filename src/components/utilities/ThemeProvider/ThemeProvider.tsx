import React from 'react';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';

export interface ThemeProps {
  children: React.ReactNode;
  theme?: {
    default?: string;
    primary?: string;
    success?: string;
    warning?: string;
    danger?: string;
    typography?: {
      fontSize: number;
      headingFontSize: number;
      paragraphLineHeight: number;
      headingLineHeight: number;
      textMargin: number;
    };
  };
}

export const defaultTheme = {
  default: '#e1e1e1',
  primary: '#409eff',
  success: '#67c23a',
  warning: '#e6a23c',
  danger: '#f56c6c',
  colors: {
    default: '#e1e1e1',
    primary: '#409eff',
    success: '#67c23a',
    warning: '#e6a23c',
    danger: '#f56c6c',
  },
  typography: {
    fontSize: 16,
    headingFontSize: 40,
    paragraphLineHeight: 1.5,
    headingLineHeight: 1.25,
    textMargin: 20,
  },
};

export const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProps) => (
  <StyledThemeProvider theme={theme}>
    <React.Fragment>{children}</React.Fragment>
  </StyledThemeProvider>
);
