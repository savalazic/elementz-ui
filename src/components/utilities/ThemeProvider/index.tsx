export { default, defaultTheme } from './ThemeProvider';
