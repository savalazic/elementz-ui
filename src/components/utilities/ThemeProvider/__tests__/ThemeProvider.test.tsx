import 'jest-dom/extend-expect';
import React from 'react';
import { render } from 'react-testing-library';

import { ThemeProvider } from '../ThemeProvider';

describe('ThemeProvider', () => {
  it('should match default snapshot', () => {
    const { container } = render(
      <ThemeProvider>
        <div>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime
            ipsam voluptatibus expedita quo iure. Asperiores mollitia maiores
            voluptatum eum voluptate!
          </p>
          <p>
            Lorem ipsum dolor sit amet consectetur, adipisicing elit. Hic, sit.
          </p>
        </div>
      </ThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
