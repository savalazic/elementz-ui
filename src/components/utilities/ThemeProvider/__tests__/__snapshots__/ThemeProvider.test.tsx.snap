// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider should match default snapshot 1`] = `
<div>
  <div>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Maxime ipsam voluptatibus expedita quo iure. Asperiores mollitia maiores voluptatum eum voluptate!
    </p>
    <p>
      Lorem ipsum dolor sit amet consectetur, adipisicing elit. Hic, sit.
    </p>
  </div>
</div>
`;
